File: 14_Longest_Commom_Prefix/Solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0){
            return "";
        }
        String common = strs[0];

        for(int i = 0; i < strs.length; ++i){
            while(strs[i].indexOf(common) != 0){
                common = common.substring(0, common.length() - 1);
                if(common == ""){
                    return "";
                }
            }
        }
        return common;
    }
}

File: 14_Longest_Commom_Prefix/Solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        common = strs[0]

        for i in range(1, len(strs)):
            while strs[i].find(common) != 0:
                common = common[0 : len(common) - 1]
                if common == "":
                    return ""
        return common